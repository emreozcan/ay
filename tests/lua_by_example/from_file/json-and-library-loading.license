https://luabyexample.techplexlabs.com/json-and-library-loading/
Lua by Example: JSON and Library Loading
by Blake Bourque

This work is licensed under a Creative Commons Attribution 3.0 Unported License.