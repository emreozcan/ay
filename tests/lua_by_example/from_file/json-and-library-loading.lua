local JSON = require 'jfjson'
local inspect = require 'inspect'

local lua_value = JSON:decode('[ "Larry", "Curly", "Moe" ]')
print('json array -> lua', inspect(lua_value))

local raw_json_text = JSON:encode({ "a", "b", "c"})
print('lua -> json array: ', raw_json_text)

local raw_json_map = JSON:encode({
    ['key'] = 'value'
})
print('lua -> json map: ', raw_json_map)