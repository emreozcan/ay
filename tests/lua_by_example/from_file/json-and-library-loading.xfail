We do not have require().
Also, this test will need the json library as well.
Maybe include it in the tests directory too?
