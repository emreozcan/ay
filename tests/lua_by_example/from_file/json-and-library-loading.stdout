json array -> lua    { "Larry", "Curly", "Moe" }
lua -> json array:     ["a","b","c"]
lua -> json map:     {"key":"value"}
